### PR TITLE
Add CLI options for simpler CLI based setup

### PR DESCRIPTION
**Pull request by miguelbalparda**
While trying to setup Shopware via CLI, I couldn't find a non interactive option to automate the install.
Sample use and output:
```
[a19b66ef@cloudhost-505127 shopware]$ APP_URL=http://test.test bin/console system:setup --help
Usage:
  system:setup [options] [--] [<APP_ENV> [<APP_URL> [<BLUE_GREEN_DEPLOYMENT> [<DATABASE_URL>] [<DATABASE_NAME>]]]]

Arguments:
  APP_ENV                                Application environment
  APP_URL                                Application URL
  BLUE_GREEN_DEPLOYMENT                  Blue green deployment
  DATABASE_URL                           Database URL
  DATABASE_NAME                          Database name

Options:
  -f, --force                            Force setup
      --no-check-db-connection           dont check db connection
      --database-url[=DATABASE-URL]      Database dsn
      --generate-jwt-keys                Generate jwt private and public key
      --jwt-passphrase[=JWT-PASSPHRASE]  JWT private key passphrase [default: "shopware"]
      --cli[=CLI]                        CLI based install
  -h, --help                             Display this help message
  -q, --quiet                            Do not output any message
  -V, --version                          Display this application version
      --ansi                             Force ANSI output
      --no-ansi                          Disable ANSI output
  -n, --no-interaction                   Do not ask any interactive question
  -e, --env=ENV                          The Environment name. [default: "prod"]
      --no-debug                         Switches off debug mode.
  -v|vv|vvv, --verbose                   Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug

[a19b66ef@cloudhost-505127 shopware]$ APP_URL=http://test.test bin/console system:setup --cli=true prod http://local.test 1 mysql://a19b66ef_sh:FlecksDirestDosageLaddie@localhost:3306 a19b66ef_sh

Shopware setup process
======================

 This tool will setup your instance.

 ! [NOTE] Checking database credentials                                                                                 

 ! [NOTE] Preparing .env                                                                                                

 /chroot/home/a19b66ef/99b23d4159.nxcli.net/shopware/.env

SHOPWARE_ES_HOSTS="elasticsearch:9200"
SHOPWARE_ES_ENABLED="0"
SHOPWARE_ES_INDEXING_ENABLED="0"
SHOPWARE_ES_INDEX_PREFIX="sw"
SHOPWARE_HTTP_CACHE_ENABLED="1"
SHOPWARE_HTTP_DEFAULT_TTL="7200"
SHOPWARE_CDN_STRATEGY_DEFAULT="id"
BLUE_GREEN_DEPLOYMENT="1"
MAILER_URL="smtp://localhost:25?encryption=&auth_mode="
APP_ENV="prod"
APP_URL="http://local.test"
APP_SECRET="def000002d59421b8350ab19e331092908552dae512286ce43d150ed52525d76cecd9a4dc0411dbcc8c5bd21b8e730bc9788197bb15d19d62872520ccddeacb5001402a3"
INSTANCE_ID="A2diKXgZs6VJg2oDdVPaxQP3WBTjWTlK"
DATABASE_URL="mysql://a19b66ef_sh:FlecksDirestDosageLaddie@localhost:3306/a19b66ef_sh"

 ! [NOTE] Writing into /chroot/home/a19b66ef/99b23d4159.nxcli.net/shopware/.env                                         

[a19b66ef@cloudhost-505127 shopware]$ bin/console system:install --create-database --basic-setup
Prepare installation

Created database `a19b66ef_sh`
Importing base schema.sql


Get collection for identifier: "core"
migrate Migrations
 346/346 [============================] 100%

 ---------- ---------------------- 
  Action     Number of migrations  
 ---------- ---------------------- 
  Migrated   346 out of 346        
 ---------- ---------------------- 

all migrations for identifier: "core" executed
cleared the shopware cache

Get collection for identifier: "core"
migrate Migrations
 346/346 [============================] 100%

 ---------- ---------------------- 
  Action     Number of migrations  
 ---------- ---------------------- 
  Migrated   346 out of 346        
 ---------- ---------------------- 

all migrations for identifier: "core" executed
cleared the shopware cache



Start theme compilation

 ! [NOTE] Took 0,005050 seconds                                                                                         



                                                                                                                        
 [OK] User "admin" successfully created.                                                                                
                                                                                                                        



                                                                                                                        
 [OK] Sales channel has been created successfully.                                                                      
                                                                                                                        

 Access tokens:
+------------+----------------------------+
| Key        | Value                      |
+------------+----------------------------+
| Access key | SWSCYXFLD0UZNLBCTW43SG1BWA |
+------------+----------------------------+

Set "Storefront" as new theme for sales channel "Storefront"
Compiling theme 9855515ad7744230b8536f6de22dfa91 for sales channel Shopware default theme
Set "Storefront" as new theme for sales channel "Headless"
Compiling theme 9855515ad7744230b8536f6de22dfa91 for sales channel Shopware default theme

Copying files for bundle: FrameworkBundle
Copying files for bundle: MonologBundle
Copying files for bundle: SwiftmailerBundle
Copying files for bundle: SensioFrameworkExtraBundle
Copying files for bundle: TwigBundle
Copying files for bundle: EnqueueBundle
Copying files for bundle: EnqueueAdapterBundle
Copying files for bundle: Framework
Copying files for bundle: System
Copying files for bundle: Content
Copying files for bundle: Checkout
Copying files for bundle: Administration
Copying files for bundle: Storefront
Copying files for bundle: Elasticsearch
Copying files for bundle: StaticKernelPluginLoader

                                                                                                                        
 [OK] Successfully copied all bundle files                                                                              
                                                                                                                        



 // Clearing the cache for the prod environment with debug false                                                        

                                                                                                                        
 [OK] Cache for the "prod" environment (debug=false) was successfully cleared.                                          
                                   


```

### CODE REVIEW

**Comment by miguelbalparda** [comment]
This is as far as I can go with my rudimentary Shopware knowledge. Any help and/or reviews are appreciated!

**Comment by miguelbalparda** [comment]
Maybe the newly added arguments should be options instead?

File: src/Command/SystemSetupCommand.php [hunk 1]
```diff
@@ -37,6 +38,13 @@ protected function configure(): void
             ->addOption('database-url', null, InputOption::VALUE_OPTIONAL, 'Database dsn')
             ->addOption('generate-jwt-keys', null, InputOption::VALUE_NONE, 'Generate jwt private and public key')
             ->addOption('jwt-passphrase', null, InputOption::VALUE_OPTIONAL, 'JWT private key passphrase', 'shopware')
+            ->addOption('cli', null, InputOption::VALUE_OPTIONAL, 'CLI based install')
+            ->addArgument('APP_ENV', InputArgument::OPTIONAL, 'Application environment')
```
Comment (miguelbalparda) [inline src/Command/SystemSetupCommand.php]: I'm not sure if these should be arguments or options. Thoughts?

Comment (pweyck) [reply src/Command/SystemSetupCommand.php]: I would prefer options.

File: src/Command/SystemSetupCommand.php [hunk 2]
```diff
@@ -66,6 +74,21 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             return 0;
         }
 
+        if ($input->getOption('cli')) {
+       		$env['APP_ENV'] = $input->getArgument('APP_ENV');
```
Comment (pweyck) [inline src/Command/SystemSetupCommand.php]: A fallback to `$_ENV[$key]`  would be nice for each variable

**Comment by pweyck** [comment]
In general, I think this could be useful for simple deployment scripts. 👍

**Comment by miguelbalparda** [comment]
Changed arguments to options, new usage below:

```
[a19b66ef@cloudhost-505127 html]$ APP_URL=http://test.test bin/console system:setup --help
Usage:
  system:setup [options]

Options:
  -f, --force                                          Force setup
      --no-check-db-connection                         dont check db connection
      --database-url[=DATABASE-URL]                    Database dsn
      --generate-jwt-keys                              Generate jwt private and public key
      --jwt-passphrase[=JWT-PASSPHRASE]                JWT private key passphrase [default: "shopware"]
      --cli[=CLI]                                      CLI based install
      --APP_ENV[=APP_ENV]                              Application environment
      --APP_URL[=APP_URL]                              Application URL
      --BLUE_GREEN_DEPLOYMENT[=BLUE_GREEN_DEPLOYMENT]  Blue green deployment
      --DATABASE_URL[=DATABASE_URL]                    Database dsn - mysql://user:password@host:port
      --DATABASE_NAME[=DATABASE_NAME]                  Database name
  -h, --help                                           Display this help message
  -q, --quiet                                          Do not output any message
  -V, --version                                        Display this application version
      --ansi                                           Force ANSI output
      --no-ansi                                        Disable ANSI output
  -n, --no-interaction                                 Do not ask any interactive question
  -e, --env=ENV                                        The Environment name. [default: "prod"]
      --no-debug                                       Switches off debug mode.
  -v|vv|vvv, --verbose                                 Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug

[a19b66ef@cloudhost-505127 html]$ APP_URL=http://test.test bin/console system:setup --cli=true --APP_ENV=prod --APP_URL=http://local.test --BLUE_GREEN_DEPLOYMENT=1 --DATABASE_URL=mysql://a19b66ef_sh:FlecksDirestDosageLaddie@localhost:3306 --DATABASE_NAME=a19b66ef_sh

Shopware setup process
======================

 This tool will setup your instance.

 ! [NOTE] Checking database credentials                                                                                 

 ! [NOTE] Preparing .env                                                                                                

 /chroot/home/a19b66ef/99b23d4159.nxcli.net/shopware/public/.env

SHOPWARE_ES_HOSTS="elasticsearch:9200"
SHOPWARE_ES_ENABLED="0"
SHOPWARE_ES_INDEXING_ENABLED="0"
SHOPWARE_ES_INDEX_PREFIX="sw"
SHOPWARE_HTTP_CACHE_ENABLED="1"
SHOPWARE_HTTP_DEFAULT_TTL="7200"
SHOPWARE_CDN_STRATEGY_DEFAULT="id"
BLUE_GREEN_DEPLOYMENT="1"
MAILER_URL="smtp://localhost:25?encryption=&auth_mode="
APP_ENV="prod"
APP_URL="http://local.test"
APP_SECRET="def000004adebba1e52e7458ea6d04475ff9f71dc78c54f80548d78a7683badb7394e6491dacd23817f54071b005664c2c3500fe07f5d6c502fce939aef140b235bf29e0"
INSTANCE_ID="l7rwEJ2y8vgvjSpFwSB7gX9aj1Iv3qhr"
DATABASE_URL="mysql://a19b66ef_sh:FlecksDirestDosageLaddie@localhost:3306/a19b66ef_sh"

 ! [NOTE] Writing into /chroot/home/a19b66ef/99b23d4159.nxcli.net/shopware/public/.env                                  


```

**Comment by J-Rahe** [comment]
Hey @miguelbalparda, i just saw you comment on the ticket.
Your PR is currently in review.
I will come back to you if there is anything that will need to be done or once it merged.

**Comment by J-Rahe** [comment]
Hey @miguelbalparda, your PR was just merged : )

**Comment by miguelbalparda** [comment]
Amazing thanks🙏 Should I document the usage of this feature somewhere?

**Comment by J-Rahe** [comment]
I don't really think so. -n is the general non interactive option and people who use this will probably fine with an --help.

**Comment by miguelbalparda** [comment]
I haven't seen this code in 6.4.1, should I port the PR or am I missing something else here?

**Comment by J-Rahe** [comment]
Hey @miguelbalparda, i just had a look and most of it has apparently seen some changes with [NEXT-14712 - Unify system:setup command ](https://github.com/shopware/production/commit/ae4f123e077f79d76215a971b75d39ca2f58630c), the only unchanged parts seem to be around line 107.
Should you find that something is not working anymore feel free to open up a new PR (or open an issue, but that will probably take longer).

**Pull request closed without merging**